hikari/impl/rest.py

```diff
@@ -1450,12 +1452,18 @@ def _build_message_payload(  # noqa: C901- Function too complex
                     final_attachments.extend(embed_attachments)
                     serialized_embeds.append(embed_payload)
 
+        serialized_poll: undefined.UndefinedOr[data_binding.JSONObject] = undefined.UNDEFINED
+        if poll is not undefined.UNDEFINED:
+            if poll is not None:
+                serialized_poll = self._entity_factory.serialize_poll(poll)
```

Comment:
```suggestion
        body.put("poll", serialized_poll, conversion=self._entity_factory.serialize_poll)
```

And moved bellow to replace the other `body.put`